handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java

```diff
@@ -507,73 +497,74 @@ static long toBIO(PrivateKey key) throws Exception {
         if (key == null) {
             return 0;
         }
-        ByteBuf buffer = Unpooled.directBuffer();
+
+        ByteBufAllocator allocator = ByteBufAllocator.DEFAULT;
+        PemEncoded pem = PemPrivateKey.toPEM(allocator, true, key);
         try {
-            buffer.writeBytes(BEGIN_PRIVATE_KEY);
-            ByteBuf wrappedBuf = Unpooled.wrappedBuffer(key.getEncoded());
-            final ByteBuf encodedBuf;
-            try {
-                encodedBuf = Base64.encode(wrappedBuf, true);
-                try {
-                    buffer.writeBytes(encodedBuf);
-                } finally {
-                    zerooutAndRelease(encodedBuf);
-                }
-            } finally {
-                zerooutAndRelease(wrappedBuf);
-            }
-            buffer.writeBytes(END_PRIVATE_KEY);
-            return newBIO(buffer);
+            return toBIO(allocator, pem.retain());
         } finally {
-            // Zero out the buffer and so the private key it held.
-            zerooutAndRelease(buffer);
+            pem.release();
         }
     }
 
-    private static void zerooutAndRelease(ByteBuf buffer) {
-        buffer.setZero(0, buffer.capacity());
-        buffer.release();
-    }
-
     /**
      * Return the pointer to a <a href="https://www.openssl.org/docs/crypto/BIO_get_mem_ptr.html">in-memory BIO</a>
      * or {@code 0} if the {@code certChain} is {@code null}. The BIO contains the content of the {@code certChain}.
      */
     static long toBIO(X509Certificate[] certChain) throws Exception {
-        if (certChain == null) {
+        if (certChain == null || certChain.length == 0) {
```

Comment:
is it correct to return 0 if the length is 0 ? If so we need to update the javadocs.